src/Tizen.NUI.Components/Attributes/ViewAttributes.cs

```diff
@@ -264,5 +280,89 @@ public override Attributes Clone()
             return new ViewAttributes(this);
         }
 
+        /// <summary>
+        /// Apply ViewAttributes to View.
+        /// </summary>
+        public virtual void ApplyToView(View view, ControlStates state = ControlStates.Normal)
+        {
+            if (view == null)
+            {
+                return;
+            }
+
+            if (Position != null)
+            {
+                view.Position = Position;
+            }
+            if (Size != null)
+            {
+                view.Size = Size;
+            }
+            if (MinimumSize != null)
+            {
+                view.MinimumSize = MinimumSize;
+            }
+            if (PositionUsesPivotPoint != null)
+            {
+                view.PositionUsesPivotPoint = PositionUsesPivotPoint.Value;
+            }
+            if (ParentOrigin != null)
+            {
+                view.ParentOrigin = ParentOrigin;
+            }
+            if (PivotPoint != null)
+            {
+                view.PivotPoint = PivotPoint;
+            }
+            if (WidthResizePolicy != null)
+            {
+                view.WidthResizePolicy = WidthResizePolicy.Value;
+            }
+            if (HeightResizePolicy != null)
+            {
+                view.HeightResizePolicy = HeightResizePolicy.Value;
+            }
+            if (SizeModeFactor != null)
+            {
+                view.SizeModeFactor = SizeModeFactor;
+            }
+
+            if (!appliedViews.ContainsKey(view))
+            {
+                appliedViews.Add(view, 0);
+                RegisterStateChangeCallback(view);
+            }
+        }
+
+        protected virtual void RegisterStateChangeCallback(View view)
+        {
+            view.FocusGained += (object sender, global::System.EventArgs e) =>
+            {
+                ApplyToView(view, ControlStates.Focused);
+            };
+
+            view.FocusLost += (object sender, global::System.EventArgs e) =>
+            {
+                ApplyToView(view, ControlStates.Normal);
+            };
+        }
+
+        public void DisApplyToView(View view)
+        {
+            if (appliedViews.ContainsKey(view))
+            {
+                appliedViews.Remove(view);
+            }
+        }
+
+        private void OnThemeChangedEvent(object sender, StyleManager.ThemeChangeEventArgs e)
+        {
+            foreach (KeyValuePair<View, int> keyValuePair in appliedViews)
+            {
+                ApplyToView(keyValuePair.Key);
+            }
+        }
+
+        private Dictionary<View, int> appliedViews = new Dictionary<View, int>();
```

Comment:
Why is a Dictionary used here instead of a List,  does the value (int) useful?  I only see the value set to 0 in all cases.
List has a Contains()  method, which you could use instead of ContainsKey().

Reply:
The Dictionary will be replaced by HashSet.

Reply:
Can you check the HashSet doesn't prevent a View from being Disposed by keeping a handle.
If so you might want to use WeakHandles.